superset/assets/src/dashboard/actions/dashboardState.js

```diff
@@ -134,19 +135,18 @@ export function saveDashboardRequestSuccess() {
 export function saveDashboardRequest(data, id, saveType) {
   const path = saveType === SAVE_TYPE_OVERWRITE ? 'save_dash' : 'copy_dash';
 
-  return dispatch =>
-    SupersetClient.post({
+  return dispatch => {
+    dispatch({ type: UPDATE_COMPONENTS_PARENTS_LIST });
+
+    return SupersetClient.post({
       endpoint: `/superset/${path}/${id}/`,
       postPayload: { data },
     })
-      .then(response =>
-        Promise.all([
-          dispatch(saveDashboardRequestSuccess()),
-          dispatch(
-            addSuccessToast(t('This dashboard was saved successfully.')),
-          ),
-        ]).then(() => Promise.resolve(response)),
-      )
+      .then(response => {
+        dispatch(saveDashboardRequestSuccess());
+        dispatch(addSuccessToast(t('This dashboard was saved successfully.')));
+        return response;
```

Comment:
typically the assumption with redux actions is that you can continue to chain promises onto them. are you certain that returning a `response` doesn't break this promise chain? that's why it was written the way it was.

Reply:
Yes, tested dashboard save flow.